pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -467,17 +467,17 @@ cdef class _Timestamp(ABCTimestamp):
             freqstr = None
 
         val = self._maybe_convert_value_to_local()
-        out = get_start_end_field(np.array([val], dtype=np.int64),
+        out = get_start_end_field(<int64_t[:1]>&val,
```

Comment:
Hmm I'm not sure about this - wouldn't this fail if the size of a pointer was not the same as the size of an `int64_t` on?

Reply:
if that were the case, wouldnt one of the 32bit builds have failed?

Reply:
A cast is going to prevent compile time complaints so no failure expected there, but the question is really about runtime

I don't think this is a safe construct, so would rather not do this unless this is a huge benefit

Reply:
> unless this is a huge benefit

not especially big, no.  closing.

Reply:
i think this change was fine, we do this kind of casting all over the place I am not sure what

> I don't think this is a safe construct, so would rather not do this unless this is a huge benefit
means

Reply:
Where else do we cast like this? I don't see any instances where we cast the address of something, particularly a scalar to an array

The size of a pointer to an int64 and the size of an int64 are not guaranteed to be the same on platforms, hence why I don't think this is well defined behavior

Reply:
> we do this kind of casting all over the place

I don't think this is correct.  I stumbled on this particular type of casting a few days ago, am pretty sure i would have noticed it if it was common in our code.

That said, if we assuage Will's concerns and this _is_ safe, then using this pattern elsewhere will allow us to share code between Timestamp/DTA, Timedelta/TDA, Period/PA .